core/primitives-core/src/serialize.rs

```diff
@@ -46,6 +46,107 @@ pub trait BaseDecode:
     }
 }
 
+/// Serialize `Vec<u8>` as `String`.
+///
+/// Fails during serialisation if the buffer does not contain valid UTF-8
+/// string.
+pub mod bytes_as_str {
```

Comment:
filed https://github.com/near/nearcore/issues/7203 to get rid of this altogether, seems possible.